components/script/dom/bindings/structuredclone.rs

```diff
@@ -111,7 +111,12 @@ unsafe fn read_blob(cx: *mut JSContext,
     let type_str = structured_reader.read_str();
     let target_global = GlobalScope::from_context(cx);
     let blob = Blob::new(&target_global, BlobImpl::new_from_bytes(blob_buffer), type_str);
-    return blob.reflector().get_jsobject().get()
+    let js_object = blob.reflector().get_jsobject().get();
+    if let Some(mut holder) = target_global.structured_clone_holder() {
```

Comment:
As pointed out, this change means we leak any blob that is sent through the structured clone algorithm because it is now always reachable from the global. I propose we return to your original design, with an important modification: there's no need to use `Rc`; we should be able to create the holder on the stack in `read_clone`, since the holder does not need to exist beyond that stack frame when the cloned values have been read into the rooted `rval` value.

Reply:
Ok, thanks, I will look into it...